pkg/controller/core/clusterqueue_controller.go

```diff
@@ -291,18 +313,79 @@ func (r *ClusterQueueReconciler) SetupWithManager(mgr ctrl.Manager) error {
 		Complete(r)
 }
 
-func (r *ClusterQueueReconciler) Status(cq *kueue.ClusterQueue) (kueue.ClusterQueueStatus, error) {
+func (r *ClusterQueueReconciler) UpdateClusterQueueCondition(
```

Comment:
Can we reuse the logic that we use for other objects that have metav1.Conditions?

Reply:
Sure.

Reply:
Done.